### PR TITLE
various: use qt5's mkDerivation

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Things done
Made sure the binaries at $out/bin were wrappers.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by worldofpeace** [comment]
Will be backported to 19.09 and 20.03.

**Review by mgttlinger — APPROVED** [review]
LGTM

**Review by acowley — APPROVED** [review]

File: pkgs/applications/graphics/qview/default.nix [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
     owner = "jurplel";
     repo = "qView";
     rev = version;
     sha256 = "15a91bs3wcqhgf76wzigbn10hayg628j84pq4j2vaxar94ak0vk7";
   };
+
   nativeBuildInputs = [ qmake ];
+
+  buildInputs = [
+    qtbase
```
[the remark this answers is not in the document]
Comment (worldofpeace) [reply pkgs/applications/graphics/qview/default.nix]: I prefer to have all lists like to have one item per line. I just forgot to do the above one.

**Comment by worldofpeace** [comment]
So I was looking through the git history, and I noticed that I already had this fixed on master and in 19.09 for seafile-client in https://github.com/NixOS/nixpkgs/pull/67270. I tried backporting the change to 19.09 and got very confused :rofl: This explains why this regressed https://github.com/NixOS/nixpkgs/issues/80976#issuecomment-590574279. Apparently the changes got reverted in 20.03 and master from https://github.com/NixOS/nixpkgs/commit/94e35fba8e1c888f03754f458006c5233a584414. @Infinisil Please be sure to use the qt version of the deriver in qt apps.


Backported to 20.03 in:
7e9bdd9743e1abf0155ac3e4196f312fea8412bd
d343ab805d9f6a78b213e32193e8d978cac221a9
309dee986c4a07768740ad1823bf0565e97ad05f
2424e5ff16e49d714967800522ce7bc51d66c436

Backported to 19.09 in:
❌️ https://github.com/NixOS/nixpkgs/pull/80982/commits/43bb5bffda01e7a6e7c18d24f755b729b43b957a
edaeeb2db671da1dfcc38979a242a101249c8d02
d8ef510fad9d4437f7cc57486216005a413689ee
7213a447173a6351398b7af20dc38f4216e309e2

**Comment by Infinisil** [comment]
Whoops sorry, will keep an eye out for that!

**Pull request merged**